### DISCUSSION TITLE
annotations in GFF3 not showing up

### DISCUSSION THREAD

**Discussion by thugib** [GMOD/jbrowse-components#3263] [General]
I have successfully added a genome file and index and it shows up in the desktop version of JBrowse2. However, I sorted a GFF3 file with GFF3_sort and indexed that with tabix and I can add it into JBrowse, but I do not see any annotation on the added track. The file seems to go into JBrowse. Here is the resulting config file:   file:///home/matthew/misclabwork/GFF3.txt
The genome fasta file has chromosomes named such as : chr1, and the GFF3 also has lower case chromosome names such as chr1, chr2, etc., so it should not be due to a mismatch of names. I cannot figure out what the problem is.

**Comment by cmdcolin** [comment]
your procedure looks right, and the mismatching names is a common source of tracks not showing up, but as you said that's probably not the issue if they are the same. can you paste your GFF or a snippet of it? and FASTA if relevant?

**Reply by thugib** [reply]
Thank you for your reply. Here is the first 4 lines of the fasta genome file:
```
>chr1
CCCTAAACCCTAAACCCTAAACCCTAAACCTCTGAATCCTTAATCCCTAAATCCCTAAATCTTTAAATCCTACATCCAT
GAATCCCTAAATACCTAATTCCCTAAACCCGAAACCGGTTTCTCTGGTTGAAAATCATTGTGTATATAATGATAATTTT
ATCGTTTTTATGTAATTGCTTATTGTTGTGTGTAGATTTTTTAAAAATATCATTTGAGGTCAATACAAATCCTATTTCT
```

And here is the first 8 lines of the GFF3 file:
```
chr1	TAIR10	chromosome	1	30427671	.	.	.	ID=chr1;Name=chr1
chr1	TAIR10	gene	3631	5899	.	+	.	ID=AT1G01010;Note=protein_coding_gene;Name=AT1G01010
chr1	TAIR10	mRNA	3631	5899	.	+	.	ID=AT1G01010.1;Parent=AT1G01010;Name=AT1G01010.1;Index=1
chr1	TAIR10	five_prime_UTR	3631	3759	.	+	.	Parent=AT1G01010.1
chr1	TAIR10	exon	3631	3913	.	+	.	Parent=AT1G01010.1
chr1	TAIR10	protein	3760	5630	.	+	.	ID=AT1G01010.1-Protein;Name=AT1G01010.1;Derives_from=AT1G01010.1
chr1	TAIR10	CDS	3760	3913	.	+	0	Parent=AT1G01010.1,AT1G01010.1-Protein;
chr1	TAIR10	exon	3996	4276	.	+	.	Parent=AT1G01010.1
```

**Comment by cmdcolin** [comment]
as far as I can tell, this should show up. which jbrowse version are you using, and you are using desktop correct? you are creating a gff and tabix file, and loading both? if you want to send me your gff to debug, let me know, can do email or whatever is easiest

**Reply by cmdcolin** [reply]
screenshot
![Screenshot from 2022-10-12 00-02-41](https://user-images.githubusercontent.com/6511937/195262414-13885d1e-d1eb-4060-8ad4-6008a8479a5b.png)

**Reply by thugib** [reply]
Thank you for your help.  
    I am using JBrowse 2 2.1.5 on Ubuntu 20.04. I downloaded GFF3 from TAIR, which is the standard site for Arabidopsis. I sorted the GFF3 with GFF_sort, then I indexed with samtools tabix, so I have both the sorted, indexed GFF3 as well as the .tbi file, and I am loading both. These files are both zipped, gz, but I can not unzip the .tbi file. 

     Here are the two files: 
[TAIR10_GFF3_gene_lc_sorted_gff.gz](https://github.com/GMOD/jbrowse-components/files/9780269/TAIR10_GFF3_gene_lc_sorted_gff.gz)
[TAIR10_GFF3_gene_lc_sorted_gff.gz.tbi.txt](https://github.com/GMOD/jbrowse-components/files/9780279/TAIR10_GFF3_gene_lc_sorted_gff.gz.tbi.txt)

    I have put a .txt suffix on the .tbi file so that I can upload it here.

I have tried using an unzipped GFF3 file and the zipped .tbi, but I still do not see any annotation on the track.

**Reply by garrettjstevens** [reply]
Are there any messages in the track area or in the developer console (open with <kbd>⌘</kbd>+<kbd>Option</kbd>+<kbd>J</kbd> or <kbd>Control</kbd>+<kbd>Shift</kbd>+<kbd>J</kbd>)? I tried briefly to open the GFF3 and could see some data but intermittently got an error like `Error: some features reference other features that do not exist in the file`.

**Reply by cmdcolin** [reply]
might relate to https://github.com/GMOD/gff-js/issues/4 https://github.com/GMOD/jbrowse/issues/1415

**Comment by cmdcolin** [comment]
@thugib we added a change in https://github.com/GMOD/jbrowse-components/pull/3272 that may help with the error "some features reference other features that do not exist in the file", can check the latest version v2.1.7 of jbrowse desktop!

**Comment by thugib** [comment]
Hi Colin,

      I tried the latest version of JBrowse dekstop 2.1.7, but still was 
not able to see the annotations.

      I have been trying to 'clean up' the gff3 file I downloaded from 
TAIR, but from using gff3_QC found in GFF3toolkit, there appear to 
hundreds of thousands errors in the file and it was not able to 'fix' 
the file. Would you be able to direct me to the Arabidopsis GFF file you 
are using ?

Matthew


On 10/20/22 12:35 PM, Colin Diesh wrote:
>
>         External Email - Use Caution
>
> @thugib 
> <https://secure-web.cisco.com/11X9svd7j67Y8lpbVUF_7UNj0oMGGE0_GV5l_qe4VVEnUEo8ZfkGpJ_oCl8blxZCFHhVrIpwzdZIz4VRaVqK3Vp1q9ANGoJ5K3qTKtzdFYOOSi5oTHZVkoBdQCt7mEcGjB05OuwyLApMOhwyeN_Q6ttk1ygrYuG5u_kw8RQBK6lbZGG0wHwr2v9VuGKG8kYutM7dgfJDzN9KPvpMTls4dW5Irhegmcw76_SVMTOeDKRqAvBGS4MNWOVf9AKCx_darwewUSCQ9VfEpMysGnxBvj9NJ1rNF0r3xJ8TXAq3jvMC4ZtbRvHJYv-XYEzoJvioZCNjUv8PiqVXvXYfGAPACDw/https%3A%2F%2Fgithub.com%2Fthugib> 
> we added a change in #3272 
> <https://secure-web.cisco.com/1OZom2_YaF0bHqF--Yjz-yhet2sOB3GS_lLSGhRkmS3ipRYZNKcLUpA8Ky09Mb3J21nspGC-lq3oTCLNoHIM_gl2RbOEHH7FbhW3Y6ccMxd7QdAfNGJAF-j-ZIwqH0jakXRFfYlsYARVLmXNYdQVI25BMgcVPUPZ5eoUYaB6GkHI2hk_ZV2rTKCpQ68MHPhCD7CNwpVrYpPyKA4rXK3aBuAwpwIBlnPeoW5uP6Z2WNjUV9PwQraJ7uWImy32UUwl2QDOW97DJoxLOEH6JKl_8BL2lAcbZgUJ0CZ2Nsa2EQFKeFN7rCXMiqRI_q34RHxzsyR_GEm0EDi34iWnbZEiAIQ/https%3A%2F%2Fgithub.com%2FGMOD%2Fjbrowse-components%2Fpull%2F3272> 
> that may help with the error "some features reference other features 
> that do not exist in the file", can check the latest version v2.1.7 of 
> jbrowse desktop!
>
> —
> Reply to this email directly, view it on GitHub 
> <https://secure-web.cisco.com/15qu9G4W-e080GejH2I8FkeqLc3tJltU8z4TYtI2eI7_51FcyTH_tqhkt-2SKvZu5EtK0SDLa8tU0B1D-m3Vfg7FOhjsIbfmRcY_DzVpKF0gjo0lz-kvKxngem92-be5telWenjajws7o80hVoS2t1_k2TOlgGPQ3RQweIJIzbS0HW4-Er2z8l1ooAt3Zrme_iWpdzAYMMX_C8U5N9YAqeKMXf4AjnMw3b2dQirRXGNCu9DGqb8yu0x-lHpcdwXp1wvgwzYU6zrufpO9DIPu5SyDuMXE9YrFGhXF-mv4D3JPrui5orAbofmaD-zZSu-W4BcSnVZfkUcqfHm36e-m6RA/https%3A%2F%2Fgithub.com%2FGMOD%2Fjbrowse-components%2Fdiscussions%2F3263%23discussioncomment-3926549>, 
> or unsubscribe 
> <https://secure-web.cisco.com/1leAdQ8SQOInTfH5_dlg5mK7KGVz0DHS-5fJhSIK_P7ukb-S2PNGJS2VkseMKDuGAKGL7bEQgW9ynyVavxid_Y5RVHq8aVnAMCvpE5u9zoC2eNsShBK0FrLdfRb6xNnl_HHQacVJj5eFtWOVAMHFMgdxtVOWP40U0dQCENvdTDAEVIGsOXha6PlHubQN8-lrQfZ0FKrdgFcam0MB6mft_ozOlK31i5w9tgx1Xcs9rF7oTRFaJd3cuXjZngQJwhfhWAZuozzY-sXDdwkb_USXuSI_mx020IIGU0JvIGFp6ggqoATrq6_HTQ3h9EsqYyCbrgADYmdt5Bi_08DpCFMtW5w/https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAB53TY7NOZXODVOAUHGV5DWEFYFZANCNFSM6AAAAAARB5VYL4>.
> You are receiving this because you were mentioned.Message ID: 
> ***@***.***>
>
The information in this e-mail is intended only for the person to whom it is addressed.  If you believe this e-mail was sent to you in error and the e-mail contains patient information, please contact the Mass General Brigham Compliance HelpLine at https://www.massgeneralbrigham.org/complianceline <https://www.massgeneralbrigham.org/complianceline> .
Please note that this e-mail is not secure (encrypted).  If you do not wish to continue communication over unencrypted e-mail, please notify the sender of this message immediately.  Continuing to send or respond to e-mail after receiving this message means you understand and accept this risk and wish to continue to communicate over unencrypted e-mail.

**Reply by cmdcolin** [reply]
I tried out wget ftp://ftp.arabidopsis.org/home/tair/Genes/TAIR10_genome_release/TAIR10_gff3/TAIR10_GFF3_genes.gff note that this file uses "chr1" so the FASTA file you have should also use this probably, or else refnamealiases will need to be configured
